gooddata-sdk/gooddata_sdk/catalog/workspace/service.py

```diff
@@ -806,6 +808,101 @@ def set_translated_texts(
                         if "description" in section["header"]:
                             section["header"]["description"] = translated.get(section["header"]["description"])
 
+    @staticmethod
+    def _add_target_tags(xliff_content: str, translate_func: Optional[Callable]) -> bytes:
+        tree = ET.ElementTree(ET.fromstring(xliff_content))
+        root = tree.getroot()
+        ns = {"ns": root.tag.split("}")[0].strip("{")}
+
+        ET.register_namespace("", ns["ns"])
+        tree = ET.ElementTree(ET.fromstring(xliff_content))
+        root = tree.getroot()
+
+        # Segment is always parent of source/target - no need to find parents
+        for segment in root.findall(".//ns:segment", namespaces=ns):
+            source = segment.find("ns:source", namespaces=ns)
+            if source is not None and "".join(source.itertext()).strip():
+                target = segment.find("ns:target", namespaces=ns)
+                if target is None:
+                    target = ET.Element("target")
+                    segment.append(target)
+                if not target.text or not target.text.strip():
+                    target_text = translate_func("".join(source.itertext()).strip() if source is not None else "")
+                    target.text = target_text
+                else:
+                    old_translation = "".join(target.itertext()).strip() if target is not None else ""
+                    target_text = translate_func(
+                        to_translate="".join(source.itertext()).strip() if source is not None else "",
+                        already_translated=True,
+                        old_translation=old_translation,
+                    )
+                    target.text = target_text
+
+        return ET.tostring(root, encoding="utf-8", xml_declaration=True)
+
+    def add_metadata_locale(
+        self,
+        workspace_id: str,
+        target_language: str,
+        translator_func: Optional[Callable] = None,
+        set_locale: bool = True,
+    ) -> None:
+        ans = self._actions_api.retrieve_translations(
+            workspace_id=workspace_id,
+            locale_request=LocaleRequest(locale=target_language),
+            _preload_content=False,
+            async_req=False,
+        )
+
+        translated_xliff = self._add_target_tags(ans.data.decode(), translator_func)
+
+        self._client.do_request(
+            method=self._client.HttpMethod.POST,
+            endpoint=f"api/v1/actions/workspaces/{workspace_id}/translations/set",
+            content_type="application/xml",
+            data=translated_xliff,
+        )
+        if set_locale:
```

Comment:
Should this happen after or before you POST new translations?
What happens if it is turned off and you POST new translations? Does it work?

Reply:
The order of the calls is irellevant. If you call it beforehand and there is no translation, it shows "en-US", but if there is some translation and it could be incomplete, it could show weird results, so to stay consistent, I'd put it after.